src/xclim/indices/_threshold.py

```diff
@@ -846,6 +848,65 @@ def maximum_consecutive_wet_days(
     return mcwd
 
 
+@declare_units(tasmax="[temperature]", tasmin="[temperature]", tas="[temperature]", thresh="[temperature]")
+def cooling_degree_days_approximation(
+    tasmax: xarray.DataArray,
+    tasmin: xarray.DataArray,
+    tas: xarray.DataArray,
+    thresh: Quantified = "18 degC",
+    freq: str = "YS",
+) -> xarray.DataArray:
+    """
+    Cooling degree days approximation.
+
+    A more robust approximation of cooling degree days as a function of the daily cycle of temperature.
+
+    Parameters
+    ----------
+    tasmax : xarray.DataArray
+        Maximum daily temperature.
+    tasmin : xarray.DataArray
+        Minimum daily temperature.
+    tas : xarray.DataArray
+        Mean daily temperature.
+    thresh : Quantified
+        Temperature threshold above which air is cooled.
+    freq : str
+        Resampling frequency.
+
+    Returns
+    -------
+    xarray.DataArray
+        Approximation of cooling degree days.
+
+    References
+    ----------
+    :cite:cts:`spinoni_2018`
+    """
+    # Where tasmax < thresh; CDD = 0
+    # Where tas <= thresh <= tasmax; CDD = (tasmax - tasmin)/4
+    # Where tasmin < thresh <= tas; CDD = [(tasmax - thresh)/2 - (thresh - tasmin)/4]
+    # Where tasmin >= thresh; CDD = tas - thresh
+    thresh = convert_units_to(thresh, tas)
+
```

Comment:
```suggestion
    tasmax = convert_units_to(tasmax, tas)
    tasmin = convert_units_to(tasmin, tas)

```
To ensure everything has the same units!